### ISSUE TITLE
First iteration of design for bulk API

### ISSUE THREAD

**Issue by jon-betts** [hypothesis/lms#1468]
We need an API to support bulk requests. Before we start this work we should:

 * Research the existing API to see what is already in place
 * Detail what the API might look like
 * Explain decisions and alternatives taken

This doesn't need to be the final cut of this, but we should ensure we have avoided any obvious pit falls and provide some evidence for decisions taken.

# Outputs

* Enough documentation to produce lively debate, but no more
* Reasons for decisions and considered alternatives
* [ ] Design of payload structure
* [ ] Design of individual resources (user, group etc)
* [ ] Details of the general behavior on errors etc.

**Comment by jon-betts** [comment]
This has been superseded by: https://github.com/hypothesis/h/pull/5905
